cranelift/codegen/src/isa/x64/inst/emit.rs

```diff
@@ -51,16 +51,43 @@ fn reg_enc(reg: Reg) -> u8 {
     reg.get_hw_encoding()
 }
 
-// F_*: these flags describe special handling of the insn to be generated.  Be
-// careful with these.  It is easy to create nonsensical combinations.
-const F_NONE: u32 = 0;
+/// A small bit field to record a REX prefix specification:
+/// - bit 0 set to 1 indicates REX.W must be 0 (cleared).
+/// - bit 1 set to 1 indicates the REX prefix must always be emitted.
+#[repr(transparent)]
+#[derive(Clone, Copy)]
+struct Rex(u8);
```

Comment:
Can this be called `RexFlags`?  Calling it simply `Rex` might lead one to believe that it itself is a Rex prefix, when it is not.

Reply:
Sure!